libp2p/Host.h

```diff
@@ -343,6 +343,11 @@ class Host: public Worker
     /// Stop registered capabilities, typically done when the network is being shut down.
     void stopCapabilities();
 
+    std::shared_ptr<Peer> peer(NodeID const& _n) const;
+
+    /// Set a handshake failure reason for a peer
+    void handshakeFailed(NodeID const& _n, HandshakeFailureReason _r);
```

Comment:
Maybe better `onHandshakeFailed`

Reply:
Also I would make it public and declare close to `startPeerSession`

Reply:
Why make this public, does it make sense to expose the concept of a handshake to consumers of Host?

Reply:
Well it looks to me like a callback similar to `startPeerSession`, the callback called by `RLPXHandshake` when handshake is finished. One is for success another one is for failure.

It works as private, because `RLPXHandshake` is a friend of `Host`, but idealluy should get rid of this friend declarations at some point.

Exposing it to the clients of `Host` is of course not great, but the proper way to deal with it could be to create a separate interface with these callbacks only, don't expose it to `Host` clients, but pass it only to `RLPXHandshake`. That's a bit complicated, at least it's not for this PR.

(In other words, we won't make it much worse, because `startPeerSession` is already public)

Reply:
> 
> 
> Well it looks to me like a callback similar to `startPeerSession`, the callback called by `RLPXHandshake` when handshake is finished. One is for success another one is for failure.
> 
> It works as private, because `RLPXHandshake` is a friend of `Host`, but idealluy we should get rid of this friend declarations at some point.
> 
> Exposing it to the clients of `Host` is of course not great, but the proper way to deal with it could be to create a separate interface with these callbacks only, don't expose it to `Host` clients, but pass it only to `RLPXHandshake`. That's a bit complicated change, at least it's not for this PR.
> 
> (In other words, we won't make it much worse, because `startPeerSession` is already public)

Ah that makes sense, thank you for clarifying! 😄 I'll make the change before merging.